modules/aggregation-layers/src/aggregation-layer-v9/gpu-aggregator/gpu-aggregator.ts

```diff
@@ -0,0 +1,253 @@
+import {ModelProps} from '@luma.gl/engine';
+import {WebGLBinSorter} from './webgl-bin-sorter';
+import {WebGLAggregationTransform} from './webgl-aggregation-transform';
+import {_deepEqual as deepEqual, log, BinaryAttribute} from '@deck.gl/core';
+
+import type {Aggregator, AggregationProps, AggregatedBin} from '../aggregator';
+import type {Device, Buffer, BufferLayout, TypedArray} from '@luma.gl/core';
+import type {ShaderModule} from '@luma.gl/shadertools';
+
+/** Options used to construct a new GPUAggregator */
+export type GPUAggregatorOptions = {
+  /** Size of bin IDs */
+  dimensions: 1 | 2;
+  /** How many properties to perform aggregation on */
+  numChannels: 1 | 2 | 3;
+  /** Buffer layout for input attributes */
+  bufferLayout?: BufferLayout[];
+  /** Define a shader function with one of the signatures
+   *  `void getBin(out int binId)`: if dimensions=1
+   *  `void getBin(out ivec2 binId)`: if dimensions=2
+   * And a shader function with one of the signatures
+   *  `void getValue(out float value)`: if numChannels=1
+   *  `void getValue(out vec2 value)`: if numChannels=2
+   *  `void getValue(out vec3 value)`: if numChannels=3
+   */
+  vs: string;
+  /** Shader modules
+   * Required to support certain layer extensions (e.g. data filter)
+   */
+  modules?: ShaderModule[];
+  /** Shadertool module defines */
+  defines?: Record<string, string | number | boolean>;
+};
+
+/** Props used to run GPU aggregation, can be changed at any time */
+export type GPUAggregationProps = AggregationProps & {
+  /** Limits of binId defined for each dimension. Ids outside of the [start, end) are ignored.
+   */
+  binIdRange: [start: number, end: number][];
+  /** Context props passed to the shader modules */
+  moduleSettings?: ModelProps['moduleSettings'];
+};
+
+/** An Aggregator implementation that calculates aggregation on the GPU */
+export class GPUAggregator implements Aggregator {
```

Comment:
Given that GPU aggregation looks [very different in WebGPU](https://webgpufundamentals.org/webgpu/lessons/webgpu-compute-shaders-histogram.html) ... do we see GPUAggregator as generalizing over WebGL and WebGPU? Or should there be separate WebGLAggregator vs. WebGPUAggregator classes? 

Apologies if I've missed or forgotten a prior thread on this, and please ignore this if it's going to distract from progress on this great stack of PRs. :)

Reply:
The plan is to have a separate class for WebGPU. I agree this should be called WebGLAggregator.